web.go

```diff
@@ -330,3 +336,78 @@ func expvarHandler(w http.ResponseWriter, r *http.Request) {
 	})
 	fmt.Fprintf(w, "\n}\n")
 }
+
+type connStats struct {
+	MaxConn   int64 `json:"max_conn"`
+	TotalConn int64 `json:"total_conn"`
+}
+
+func (provider *WebProvider) getBackendConnStatsHandler(response http.ResponseWriter, request *http.Request) {
+	vars := mux.Vars(request)
+	providerID := vars["provider"]
+	backendID := vars["backend"]
+	currentConfigurations := provider.server.currentConfigurations.Get().(configs)
+	if providerConf, ok := currentConfigurations[providerID]; ok {
+		if backendConf, ok := providerConf.Backends[backendID]; ok {
+			if connLimiter, ok := provider.server.backendConnLimits[backendID]; ok {
+				if totalConn, ok := getTotalConn(connLimiter); ok {
+					templatesRenderer.JSON(response, http.StatusOK, &connStats{
+						MaxConn:   backendConf.MaxConn.Amount,
+						TotalConn: totalConn,
+					})
+					return
+				}
+			}
+		}
+	}
+
+	http.NotFound(response, request)
+}
+
+type providers struct {
+	Backends map[string]connStats `json:"backends"`
+}
+
+func (provider *WebProvider) getConnStatsHandler(response http.ResponseWriter, request *http.Request) {
+	payload := make(map[string]providers)
+
+	currentConfigurations := provider.server.currentConfigurations.Get().(configs)
+	for p := range currentConfigurations {
+		payload[p] = providers{
+			Backends: make(map[string]connStats),
+		}
+		for b := range currentConfigurations[p].Backends {
+			if connLimiter, ok := provider.server.backendConnLimits[b]; ok {
+				if totalConn, ok := getTotalConn(connLimiter); ok {
+					payload[p].Backends[b] = connStats{
+						MaxConn:   currentConfigurations[p].Backends[b].MaxConn.Amount,
+						TotalConn: totalConn,
+					}
+				}
+			}
+		}
+	}
+
+	templatesRenderer.JSON(response, http.StatusOK, payload)
+}
+
+func getTotalConn(cl *connlimit.ConnLimiter) (int64, bool) {
```

Comment:
Indeed, I can send them a PR for that lib too, but let's first see if they're interested in it and if they want any changes in the design. Before that, I was hoping to get it running in our system to make sure everything is fine. There are loads of moving pieces...